src/gnome-utils/dialog-options.c

```diff
@@ -2420,6 +2420,149 @@ gnc_option_set_ui_widget_dateformat (GNCOption *option, GtkBox *page_box,
+    gnc_option_changed_option_cb(widget, option);
+    return;
+}
+
+
+static GtkWidget *
+gnc_option_set_ui_widget_plot_size (GNCOption *option, GtkBox *page_box,
+                                     char *name, char *documentation,
+                                     /* Return values */
+                                     GtkWidget **enclosing, gboolean *packed)
+{
+    GtkWidget *value_px, *value_percent;
+    GtkWidget *label;
+    GtkWidget *px_butt, *p_butt;
+    GtkWidget *hbox;
+    gchar *colon_name;
+    GtkAdjustment *adj_px, *adj_percent;
+    gdouble lower_bound = G_MINDOUBLE;
+    gdouble upper_bound = G_MAXDOUBLE;
+    gdouble step_size = 1.0;
+    int num_decimals = 0;
+
+    colon_name = g_strconcat(name, ":", NULL);
+    label = gtk_label_new(colon_name);
+    gtk_misc_set_alignment(GTK_MISC(label), 1.0, 0.5);
+    g_free(colon_name);
+
+    hbox = gtk_hbox_new(FALSE, 5);
+
+    *enclosing = gtk_hbox_new(FALSE, 5);
+
+    gtk_box_pack_start(GTK_BOX(*enclosing), label, FALSE, FALSE, 0);
+    gtk_box_pack_start(GTK_BOX(*enclosing), hbox, FALSE, FALSE, 0);
+
+    gnc_option_get_range_info(option, &lower_bound, &upper_bound,
+                              &num_decimals, &step_size);
+    adj_px = GTK_ADJUSTMENT(gtk_adjustment_new(lower_bound, lower_bound,
+                                            upper_bound, step_size,
+                                            step_size * 5.0,
+                                            0));
+
+    value_px = gtk_spin_button_new(adj_px, step_size, num_decimals);
+    gtk_spin_button_set_numeric(GTK_SPIN_BUTTON(value_px), TRUE);
+
+    {
+        gdouble biggest;
+        gint num_digits;
+
+        biggest = ABS(lower_bound);
+        biggest = MAX(biggest, ABS(upper_bound));
+
+        num_digits = 0;
+        while (biggest >= 1)
+        {
+            num_digits++;
+            biggest = biggest / 10;
+        }
+
+        if (num_digits == 0)
+            num_digits = 1;
+
+        num_digits += num_decimals;
+
+        gtk_entry_set_width_chars(GTK_ENTRY(value_px), num_digits);
+    }
+    gtk_spin_button_set_value(GTK_SPIN_BUTTON(value_px), (upper_bound / 2)); //default
+    g_signal_connect(G_OBJECT(value_px), "changed",
+                     G_CALLBACK(gnc_option_changed_widget_cb), option);
+
+    adj_percent = GTK_ADJUSTMENT(gtk_adjustment_new(1, 10, 100, 1, 5.0, 0));
+    value_percent = gtk_spin_button_new(adj_percent, 1, 0);
+    gtk_spin_button_set_numeric(GTK_SPIN_BUTTON(value_percent), TRUE);
+    gtk_spin_button_set_value(GTK_SPIN_BUTTON(value_percent), 100); //default
+    gtk_entry_set_width_chars(GTK_ENTRY(value_percent), 3);
+    gtk_widget_set_sensitive(value_percent, FALSE);
+
+    g_signal_connect(G_OBJECT(value_percent), "changed",
+                     G_CALLBACK(gnc_option_changed_widget_cb), option);
+
+    px_butt = gtk_radio_button_new_with_label(NULL, _("Pixels"));
+    gtk_toggle_button_set_active (GTK_TOGGLE_BUTTON(px_butt), TRUE);
```

Comment:
Heh, while that is how it has always been, I think we want the percent option to be the default. I think a full width/height chart is more often useful than that 400px by 400px. The latter is almost always too small.